### PR TITLE
3761 - 3

### PR DESCRIPTION
**Pull request by LuginaJulia**
## Номер
3761

## Номер задания
3

## Ссылка на видео с демо
https://drive.google.com/open?id=1tTbfWp9oz73DmfNDMXK28ZfFMjS4JHy4

### CODE REVIEW

File: 3761/3/app/controllers/application_controller.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+require 'rack-flash'
+
+class ApplicationController < Sinatra::Base
+  register Sinatra::Session
+  register Sinatra::ActiveRecordExtension
+
+  use Rack::Flash
+  configure do
+    set :session_fail, '/sign_up'
+    set :session_secret, 'DcdvRSo0perSeKr3t!'
```
Comment (anatoliliotych) [inline 3761/3/app/controllers/application_controller.rb]: Bad idea to store some secrets in repo. Please, move to .env file

File: 3761/3/app/controllers/place_controller.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  end
+
+  attr_reader :review
+
+  private
+
+  def average_score
+    review.inject(0) do |score, review|
+      score + review.grade
+    end / review.count
```
Comment (anatoliliotych) [inline 3761/3/app/controllers/place_controller.rb]: it's unreadable. try to not use such style

File: 3761/3/db/migrate/20190725084050_create_users_table.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+# :reek:FeatureEnvy
+class CreateUsersTable < ActiveRecord::Migration[5.2]
+  def change
+    create_table :users do |column|
+      column.string :name
```
Comment (anatoliliotych) [inline 3761/3/db/migrate/20190725084050_create_users_table.rb]: If you add some validations on model level, try to keep them on db columns too.

File: 3761/3/app/controllers/user_controller.rb [hunk 4]
```diff
@@ -0,0 +1,51 @@
+class UserController < ApplicationController
+  get '/sign_in' do
+    if session?
```
Comment (Moleved) [inline 3761/3/app/controllers/user_controller.rb]: Try
```ruby
redirect '/' && return if session?

erb :sign_up
```

File: 3761/3/app/controllers/user_controller.rb [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+      redirect '/'
+    else
+      erb :sign_in
+    end
+  end
+
+  post '/sign_in' do
+    @user = User.find_by(email: params['email'])
```
Comment (Moleved) [inline 3761/3/app/controllers/user_controller.rb]: May be separate method

File: 3761/3/app/controllers/user_controller.rb [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+  end
+
+  post '/sign_up' do
+    @user = User.new(name: params['name'], email: params['email'])
+    @user.password = params[:password]
+    unless @user.valid?
+      flash[:message] = I18n.t(:account_already_exist)
+      redirect '/sign_up'
+    end
+    unless Truemail.valid?(params['email'])
```
Comment (Moleved) [inline 3761/3/app/controllers/user_controller.rb]: Add validation to model

File: 3761/3/app/controllers/application_controller.rb [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+require_relative '../helpers/session_helper'
+
+class ApplicationController < Sinatra::Base
+  register Sinatra::Session
+  register Sinatra::ActiveRecordExtension
+  helpers UserHelper
+  helpers SessionHelper
+
+  Dotenv.load
+  SESSION_SECRET = ENV['SESSION_SECRET']
```
Comment (anatoliliotych) [inline 3761/3/app/controllers/application_controller.rb]: what is the reason to set constant? we use it only once during configuration, just put it as is

Comment (anatoliliotych) [reply 3761/3/app/controllers/application_controller.rb]: And where is .env.sample?

**Pull request merged**